patches/tModLoader/Terraria/Main.TML.cs

```diff
@@ -447,4 +451,51 @@ public static void AddSignalTraps()
 		SIGINTHandler = PosixSignalRegistration.Create(PosixSignal.SIGINT, Handle);
 		SIGTERMHandler = PosixSignalRegistration.Create(PosixSignal.SIGTERM, Handle);
 	}
+
+	private static string newsText = "Checking...";
+	private static string newsURL = null;
+	private static bool newsChecked = false;
+	private static bool newsIsNew = false;
+	private static void HandleNews(Color menuColor)
+	{
+		if (menuMode == 0) {
+			if (!newsChecked) {
+				newsChecked = true;
+				// Download latest news, save to config.json.
+				// https://partner.steamgames.com/doc/webapi/ISteamNews
+				client.GetStringAsync("https://api.steampowered.com/ISteamNews/GetNewsForApp/v2/?appid=1281930&count=1").ContinueWith(response => {
+					JObject o = JObject.Parse(response.Result);
```

Comment:
Consider error handling in the event computer / steam host is offline to minimize logged text.

Consider error handling for various steam responses: https://partner.steamgames.com/doc/webapi_overview/responses#status_codes